lib/plugins/aws/invokeLocal/index.js

```diff
@@ -443,113 +417,109 @@ class AwsInvokeLocal {
       'invokeLocal',
       'artifact'
     );
-    return decompress(artifact, destination).then(() => destination);
+    await decompress(artifact, destination);
+    return destination;
   }
 
   getEnvVarsFromOptions() {
-    const envVarsFromOptions = {};
     // Get the env vars from command line options in the form of --env KEY=value
-    [].concat(this.options.env || []).forEach(itm => {
-      const splitItm = itm.split('=');
-      envVarsFromOptions[splitItm[0]] = splitItm.slice(1, splitItm.length).join('=') || '';
-    });
+    const envVarsFromOptions = {};
+    for (const envOption of [].concat(this.options.env || [])) {
+      const [varName, ...varRest] = envOption.split('=');
+      const varValue = varRest.join('=');
+      envVarsFromOptions[varName] = varValue;
+    }
     return envVarsFromOptions;
   }
 
-  ensurePackage() {
-    if (this.options['skip-package']) {
-      return fse
-        .access(
+  async ensurePackage() {
+    try {
```

Comment:
Done!

It's now a bit uglier because we need to duplicate the `spawn` but it's exactly the same logic.

Reply:
> It's now a bit uglier because we need to duplicate the spawn but it's exactly the same logic.

No, there was no need to duplicate it. Your approach was perfect in terms that you didn't duplicate `spawn`, and it'll be great to maintain that.

My point was simply, that `try/catch` should be within `if` and not wrapping it

Reply:
🤦‍♂️

So much back and forth that I'm starting to miss the obvious....

Done!